rasa_core/server.py

```diff
@@ -288,6 +290,50 @@ def parse(self, request, sender_id):
                          "parse: {}".format(e), exc_info=1)
             return json.dumps({"error": "{}".format(e)})
 
+    @app.route("/conversations/<sender_id>/respond",
+               methods=['GET', 'POST', 'OPTIONS'])
+    @check_cors
+    @requires_auth
+    @ensure_loaded_agent
+    def respond(self, request, sender_id):
+        request.setHeader('Content-Type', 'application/json')
+        if request.method.decode('utf-8', 'strict') == 'GET':
+            request_params = {
+                key.decode('utf-8', 'strict'): value[0].decode('utf-8',
+                                                               'strict')
+                for key, value in request.args.items()}
+        else:
+            content = request.content.read()
```

Comment:
same here

Reply:
maybe one function for the whole block is fine, e.g. `request_params = request_parameters(request)`